File: ChefApp/app/src/main/java/com/emedinaa/chefapp/data/socket/SocketMapper.java
package com.emedinaa.chefapp.data.socket;

import android.util.Log;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author : Eduardo Medina
 * @see : https://developer.android.com/index.html
 * @since : 8/18/18
 */
public class SocketMapper {

    private final String PARAM_SUCCESS="success";
    private final String PARAM_MESSAGE="message";

    public SocketResponse convert(JSONObject jsonObject){
        SocketResponse socketResponse= new SocketResponse();
        if(jsonObject==null)return socketResponse;

        boolean success=false;
        String message="";
        try {
            //success = jsonObject.getBoolean("success");
            //message = jsonObject.getString("message");
            success = jsonObject.getBoolean(PARAM_SUCCESS);
            message = jsonObject.getString(PARAM_MESSAGE);
        } catch (JSONException e) {
            Log.e("CONSOLE", e.getMessage());
        }
        socketResponse.setSuccess(success);
        socketResponse.setMessage(message);
        return socketResponse;
    }
}

File: RestaurantApp/app/src/main/java/com/emedinaa/kotlinapp/socket/SocketMapper.kt
package com.emedinaa.kotlinapp.socket

import android.util.Log
import org.json.JSONException
import org.json.JSONObject

class SocketMapper {

    private val PARAM_SUCCESS = "success"
    private val PARAM_MESSAGE = "message"

    fun convert(jsonObject: JSONObject?): SocketResponse {
        var socketResponse:SocketResponse? = null
        if (jsonObject == null) return SocketResponse(false,"")

        var success = false
        var message = ""
        try {
            success = jsonObject?.getBoolean(PARAM_SUCCESS)
            message = jsonObject?.getString(PARAM_MESSAGE)
        } catch (e: JSONException) {
            Log.e("CONSOLE", e.message)
        }
        socketResponse= SocketResponse(success,message)
        return socketResponse
    }
}